src/movies.js

```diff
@@ -1,16 +1,95 @@
 // Iteration 1: All directors? - Get the array of all directors.
+
+function getAllDirectors (movies){
+    let directors = movies.map(function(movie){
+        return movies.directors;
+    });
+    return directors;
+}   
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(movies){
+    let dramaMovies = movies.filter(function(movie){
+        movie.director == "Steven Spielberg" && movie.genre.includes("Drama");
```

Comment:
In these methods (map, filter, sort, etc...) you need to return something in the inner function. If this case you are evaluating the .director and .genre but you are not returning the evaluation. 
This might be confusing because when you reduce the syntax to a one line, you don't need to add the return keyword. However, when you do the map method statement in multiple lines inside curly brackets you need to add the return keyword.